internal/bminventory/inventory_test.go

```diff
@@ -15060,6 +15060,56 @@ var _ = Describe("TestRegisterCluster", func() {
 		})
 	})
 
+	Context("Authz and soft timeouts", func() {
+		var (
+			authCtx      context.Context
+			mockOcmAuthz *ocm.MockOCMAuthorization
+			payload      *ocm.AuthPayload
+			orgID1       = "300F3CE2-F122-4DA5-A845-2A4BC5956996"
+			userName1    = "test_user_1"
+		)
+
+		BeforeEach(func() {
+			db, dbName = common.PrepareTestDB()
+			bm = createInventory(db, Config{})
+			bm.clusterApi = cluster.NewManager(cluster.Config{}, common.GetTestLog().WithField("pkg", "cluster-monitor"),
+				db, commontesting.GetDummyNotificationStream(ctrl), mockEvents, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, nil, false)
+			cfg := auth.GetConfigRHSSO()
+			cfg.EnableOrgBasedFeatureGates = true
+			mockOcmAuthz = ocm.NewMockOCMAuthorization(ctrl)
+			mockOcmClient := &ocm.Client{Cache: cache.New(10*time.Minute, 30*time.Minute), Authorization: mockOcmAuthz}
+			bm.authzHandler = auth.NewAuthzHandler(cfg, mockOcmClient, common.GetTestLog().WithField("pkg", "auth"), db)
+			payload = &ocm.AuthPayload{Role: ocm.UserRole}
+			payload.Username = userName1
+			payload.Organization = orgID1
+			authCtx = context.WithValue(ctx, restapi.AuthKey, payload)
+			mockClusterRegisterSteps()
+			mockUsageReports()
+			mockAMSSubscription(authCtx)
+			mockMetric.EXPECT().ClusterRegistered().Times(1)
+			mockEvents.EXPECT().SendClusterEvent(gomock.Any(), eventstest.NewEventMatcher(
+				eventstest.WithNameMatcher(eventgen.ClusterRegistrationSucceededEventName))).Times(1)
+		})
+		DescribeTable("organizational soft timeouts", func(softTimeouts bool) {
+			mockOcmAuthz.EXPECT().CapabilityReview(gomock.Any(), userName1, ocm.SoftTimeoutsCapabilityName, ocm.OrganizationCapabilityType).Return(softTimeouts, nil).Times(1)
+
+			reply := bm.V2RegisterCluster(authCtx, installer.V2RegisterClusterParams{
+				NewClusterParams: &models.ClusterCreateParams{
+					Name:             swag.String("some-cluster-name"),
+					OpenshiftVersion: swag.String("4.12.0-someFakeFlavour"),
+					CPUArchitecture:  common.X86CPUArchitecture,
+					PullSecret:       swag.String(fakePullSecret),
+				},
+			})
+			Expect(reply).To(BeAssignableToTypeOf(installer.NewV2RegisterClusterCreated()))
+			actual := reply.(*installer.V2RegisterClusterCreated)
+			Expect(actual.Payload.OrgSoftTimeoutsEnabled).To(Equal(softTimeouts))
+		},
+			Entry("soft timeouts prohibited", false),
```

Comment:
Maybe we should test that if there is an error when we ask for capability we don't fail?

Reply:
What will we do in case of failure?

Reply:
continue, this is what is done in code, no?

Reply:
Yes, so what is the point to catch error?

Reply:
I will add error handling